mlflow/server/js/src/model-registry/components/SchemaTable.tsx

```diff
@@ -54,16 +54,44 @@ export class SchemaTableImpl extends React.PureComponent<Props> {
     );
   };
 
-  getSchemaTypeRepr = (schemaTypeSpec: any) => {
-    let { type } = schemaTypeSpec;
-    if (schemaTypeSpec.type === MODEL_SCHEMA_TENSOR_TYPE) {
-      type = `Tensor (dtype: ${schemaTypeSpec['tensor-spec'].dtype}, shape: [${schemaTypeSpec['tensor-spec'].shape}])`;
+  getSchemaTypeRepr = (schemaTypeSpec: ColumnSpec | TensorSpec): string => {
+    const { type } = schemaTypeSpec;
+
+    let repr: string = type;
+    if (schemaTypeSpec.type === 'tensor') {
+      repr = `Tensor (dtype: ${schemaTypeSpec['tensor-spec'].dtype}, shape: [${schemaTypeSpec['tensor-spec'].shape}])`;
+    } else {
+      repr = this.getColumnTypeRepr(schemaTypeSpec);
     }
 
     // If the "optional" property is present and true, wrap the type around an "Optional[]"
+    // NOTE: newer model signatures will have "required" instead of "optional", but we currently
+    // have to support both since both exist.
     if (schemaTypeSpec.optional) {
-      type = `Optional[${type}]`;
+      repr = `Optional[${repr}]`;
+    } else if (schemaTypeSpec.required) {
+      repr = `${repr} (required)`;
     }
+
+    return repr;
+  };
+
+  getColumnTypeRepr = (columnType: ColumnType): string => {
+    const { type } = columnType;
+
+    if (type === 'object') {
+      const propertyReprs = Object.keys(columnType.properties).map((propertyName) => {
+        const property = columnType.properties[propertyName];
+        const required = property.required ? ' (required)' : '';
```

Comment:
```
"content": string
```

Does this look like it's optional?

Reply:
Yeah I think we're a bit limited by the width of the text box at the moment—the `(required)` drops to the next line. As an easy fix we could try increasing the width of the "type" column. Alternatively I could write an actual component to render object properties on separate lines.

Reply:
Oh or do you mean properties that are not required are not obviously optional? I suppose we could add (optional) as well, but I worry that the signatures will get super messy.

Reply:
> do you mean properties that are not required are not obviously optional?

This is what I meant. Can we use `"content": string (optional)` if optional else `"content": string"`?

Reply:
Sounds good to me, I agree. @serena-ruan should we change the `repr` in python to be the same?

Reply:
Good idea!

Reply:
I also thought about using `?` to indicate optional, and self-rejected this idea :)

Reply:
OpenAI uses both required and optional:

<img width="654" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/6f7a8f93-94ac-43ca-9bec-c1e8037fa900">